aten/src/ATen/function_wrapper.py

```diff
@@ -92,6 +93,10 @@ def TypedDict(name, attrs, total=True):  # type: ignore
 ${return_type} ${api_name}(${type_method_formals});
 """)
 
+NATIVE_DISPATCH_DECLARATION_CONST = CodeTemplate("""\
+${return_type} ${api_name}(${type_method_formals}) const;
```

Comment:
Err... if these are actually methods on a class, shouldn't all of the methods be const?

Reply:
im not sure if im following. This affects this [piece](https://github.com/pytorch/pytorch/commit/c8dd1c341d9d8cff27d57e9efb3822a732e05411#diff-bf247f3bb498cd6e9bd69bef6931580bR538-R543).

It was const and stays const.

Reply:
Link isn't working for me. Can you paste it in here?

Reply:
OK, I think I understand better what is going on here.

In this part of the diff, you took an existing `NATIVE_DISPATCH_DECLARATION` template, saw that you needed a version of it with a `const` qualifier, and so you added another version of the template `NATIVE_DISPATCH_DECLARATION_CONST` with the template. This is a very mechanistic way to do the code change, done by looking for something similar to what you need in another context. In my experience, we end up with clearer code if we dig a little deeper into what, *semantically*, we're trying to achieve here. So let's pull back those covers.

What does the `NATIVE_DISPATCH_DECLARATION` template do, why is it called "native dispatch declaration"? This template is used to populate functions in the namespace `at::TypeDefault`, these functions contain code to forward onto the actual `native::` declarations which do all the heavy lifting. An example definition is `NATIVE_DISPATCH_DEFINITION_DEFAULT`; you can see that the template names are related by stating `DECLARATION` and `DEFINITION`.

With this context in mind, the change you've made here is really confusing. You are referring to method definitions on a class (I'm actually not sure what the class is, but I don't think the exact identity is important for this conversation). That is fine, I can believe this is something you need... but this has *nothing* to do with "native dispatch" in the sense that the original templates are used for? No more plain is this confusion than these lines of the diff:

```
            if is_factory_method:
                code = gen_native_dispatch_declaration(option)
                top_env['tensor_method_declarations'].append(code.declaration)
                top_env['tensor_method_definitions'].append(code.definition)
```

You generate a "native declaration"... and then promptly stuff it in `tensor_methods`. These don't sound like the same thing at all to me!

----

OK, so what am I asking for?

1. Rename this template. I'm not going to tell you what name to use, please think about what the template is being used for and come up with an appropriate name based on the use.
2. Rename `gen_native_dispatch_declaration` to reflect this new name.